src/System.Management.Automation/engine/PropertyAccessor.cs

```diff
@@ -56,142 +28,41 @@ internal enum PropertyScope
-        /// <para>
-        /// The default value is PSLevel.Informational
-        /// </para>
-        /// </returns>
-        internal abstract PSLevel GetLogLevel();
-
-        /// <summary>
-        /// Gets the bitmask of the PSChannel values to log.
-        /// </summary>
-        /// <returns>
-        /// A bitmask of PSChannel.Operational and/or PSChannel.Analytic
-        /// <para>
-        /// The default value is PSChannel.Operational
-        /// </para>
-        /// </returns>
-        internal abstract PSChannel GetLogChannels();
-
-        /// <summary>
-        /// Gets the bitmask of keywords to log.
-        /// </summary>
-        /// <returns>
-        /// A bitmask of PSKeyword values.
-        /// <para>
-        /// The default value is all keywords other than UseAlwaysAnalytic
-        /// </para>
-        /// </returns>
-        internal abstract PSKeyword GetLogKeywords();
-
-#endif // UNIX
-        #endregion // Interface Methods
-    }
-
-    /// <summary>
-    /// JSON configuration file accessor
-    ///
-    /// Reads from and writes to configuration files. The values stored were
-    /// originally stored in the Windows registry.
-    /// </summary>
-    internal class JsonConfigFileAccessor : ConfigPropertyAccessor
-    {
         private string psHomeConfigDirectory;
         private string appDataConfigDirectory;
         private const string configFileName = "PowerShellProperties.json";
 
         /// <summary>
-        /// Lock used to enable multiple concurrent readers and singular write locks within a
-        /// single process.
-        /// TODO: This solution only works for IO from a single process. A more robust solution is needed to enable ReaderWriterLockSlim behavior between processes.
+        /// Lock used to enable multiple concurrent readers and singular write locks within a single process.
+        /// TODO: This solution only works for IO from a single process.
+        ///       A more robust solution is needed to enable ReaderWriterLockSlim behavior between processes.
         /// </summary>
         private ReaderWriterLockSlim fileLock = new ReaderWriterLockSlim();
 
-        internal JsonConfigFileAccessor()
+        private PowerShellConfig()
         {
-            //
             // Sets the system-wide configuration directory
-            //
-            Assembly assembly = typeof(PSObject).GetTypeInfo().Assembly;
-            psHomeConfigDirectory = Path.GetDirectoryName(assembly.Location);
+            psHomeConfigDirectory = Utils.DefaultPowerShellAppBase;
 
-            //
             // Sets the per-user configuration directory
             // Note: This directory may or may not exist depending upon the
             // execution scenario. Writes will attempt to create the directory
             // if it does not already exist.
-            //
             appDataConfigDirectory = Utils.GetUserConfigurationDirectory();
```

Comment:
We can install PowerShell Core side-by-side with other PowerShell Core-s. So we should have configs for "default" system wide installation and for "local" for every user installation.
Also it seems we should have GPO option - "Force GPO for Side-By-Side PowerShell installations" - by default we should apply GPO only for system wide PowerShell installation.

Reply:
Based on internal discussion today, the current plan is to have PSCore6 GPO settings separate from Windows PowerShell.  For now, we don't plan to have side-by-side GPO settings per version of PSCore6 until there is customer need for it.  New admx will be published later for the PSCore6 settings (which will be the same as Windows PowerShell, but reside in a different location in registry).

Reply:
GPO is Enterprise feature. GPO should allow Domain Admins get full control on PowerShell. I full trust MSFT experts but I'd prefer discuss GPO in a Issue. Also we forget about AppLocker.

Reply:
@iSazonov since PSCore6 is not inbox in Windows and doesn't replace Windows PowerShell, we shouldn't use the same settings.  PSCore6 in this case is no different from any 3rd party app.